wake-lock/idlharness.https.html

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Wake Lock API IDL tests</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="help" href="https://w3c.github.io/wake-lock/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/WebIDLParser.js"></script>
+<script src="/resources/idlharness.js"></script>
+<script>
+"use strict";
+
+function doTest([dom_idl, wakelock_idl, wakelock]) {
+  const idl_array = new IdlArray();
+  idl_array.add_untested_idls(dom_idl);
+  idl_array.add_untested_idls('interface Navigator {};');
+  idl_array.add_idls(wakelock_idl);
+  self.wakelock = wakelock;
+  self.request = wakelock.createRequest();
+  idl_array.add_objects({WakeLock: ['wakelock'], WakeLockRequest: ['request']});
+  idl_array.test();
+}
+
+promise_test(() => {
+  return Promise.all([
+    fetch("/interfaces/dom.idl").then((response) => response.text()),
+    fetch("/interfaces/wake-lock.idl").then((response) => response.text()),
+    navigator.getWakeLock("screen")
```

Comment:
Will this prompt for permission? I can't remember :)

Reply:
This is up to user agent implementation, the spec didn't constrain to request user permission.

Reply:
This might cause tests to lock up tho. Maybe we need some kind of timeout here?

Reply:
(actually, the test can time itself out... ignore my last comment there.)